index.d.ts

```diff
@@ -0,0 +1,1100 @@
+// Type definitions for [~THE LIBRARY NAME~] [~OPTIONAL VERSION NUMBER~]
+// Project: [~THE PROJECT NAME~]
+// Definitions by: [~YOUR NAME~] <[~A URL FOR YOU~]>
+
+export as namespace Draggable;
+
+// type definitions
+export type DraggableContainer = HTMLElement[] | NodeList | HTMLElement
+
+type DraggableEventName = 'draggable:initialized' | 'draggable:destroy' | string
+type DragEventName = 'drag:start' | 'drag:move' | 'drag:over' | 'drag:over:container' | 'drag:out' | 'drag:out:container' | 'drag:stop' | 'drag:pressure' | string
+type SortEventName = 'sortable:start' | 'sortable:sorted' | 'sortable:stop' | string
+type MirrorEventName = 'mirror:created' | 'mirror:attached' | 'mirror:move' | 'mirror:destroy' | string
+
+// plugin
+export class DraggablePlugin {
+  draggable: Draggable
+
+  constructor(draggable: Draggable)
+
+  attach(): void
+
+  detach(): void
+
+  getOptions(): Object
+}
+
+// sensor
+export class DraggableSensor {
+
+  /**
+   * Current containers
+   * @property containers
+   * @type {HTMLElement[]}
+   */
+  containers: HTMLElement[]
+
+  /**
+   * Current options
+   * @property options
+   * @type {Object}
+   */
+  options: Object
+
+  /**
+   * Current drag state
+   * @property dragging
+   * @type {Boolean}
+   */
+  dragging: boolean
+
+  /**
+   * Current container
+   * @property currentContainer
+   * @type {HTMLElement}
+   */
+  currentContainer: HTMLElement
+
+  constructor(containers: HTMLElement[], options: Object)
+
+  attach(): DraggableSensor
+
+  detach(): DraggableSensor
+}
+
+// draggable options
+interface DraggableOptions {
+  // basic props
+  draggable?: string,
+  handle?: string,
+  delay?: string,
+  plugins?: any[],
+  sensors?: any[],
+  appendTo?: string | HTMLElement | Function,
+  classes?: Object,
```

Comment:
@tsov  is there a defined set of these? At worst it should be a dictionary type:

```
classes: {
  [key: string]: string,
}
```

so that it's limited to string key/value pairs.